--- parrot-2.0.0.orig/tools/dev/pbc_to_exe.pir
+++ parrot-2.0.0/tools/dev/pbc_to_exe.pir
@@ -4,7 +4,7 @@
 
 =head1 NAME
 
-pbc_to_exe
+pbc_to_exe - Generate executables from Parrot bytecode
 
 =head1 DESCRIPTION
 
--- parrot-2.0.0.orig/src/parrot_debugger.c
+++ parrot-2.0.0/src/parrot_debugger.c
@@ -4,7 +4,7 @@
 
 =head1 NAME
 
-parrot_debugger
+parrot_debugger - The Parrot Debugger
 
 =head1 DESCRIPTION
 
