Index: parrot-2.9.1/src/gc/system.c
===================================================================
--- parrot-2.9.1.orig/src/gc/system.c	2010-11-28 00:56:24.000000000 +0000
+++ parrot-2.9.1/src/gc/system.c	2010-11-28 00:57:57.000000000 +0000
@@ -188,7 +188,7 @@
 
 #  endif /* __hpux */
 
-        trace_mem_block(interp, base,
+        trace_mem_block(interp, mem_pools, base,
                 (size_t)current_regstore_top);
 
 #else /* !__ia64__ */
