--- tools/dev/install_files.pl	2009-07-13 10:26:40.000000000 +0200
+++ tools/dev/install_files.pl	2009-07-13 13:37:49.000000000 +0200
@@ -163,7 +163,7 @@
             # libdir as it is typically done with automake installed packages.
             # If there is a use case to make this configurable we'll add a
             # seperate --pkgconfigdir option.
-            $filehash->{DestDirs} = ['pkgconfig', $parrotdir];
+            $filehash->{DestDirs} = ['pkgconfig'];
             return($filehash);
         },
     },
--- config/gen/makefiles/parrot_pc.in	2009-06-06 16:33:32.000000000 +0200
+++ config/gen/makefiles/parrot_pc.in.new	2009-07-13 23:09:45.000000000 +0200
@@ -7,4 +7,4 @@
 Description: virtual machine to execute bytecode for interpreted languages
 Version: @VERSION@
 Libs: -L${libdir} -lparrot @icu_shared@ @libs@
-Cflags: -I${includedir}
+Cflags: -I${includedir}@versiondir@
--- lib/Parrot/Install.pm	2009-06-01 09:29:57.000000000 +0200
+++ lib/Parrot/Install.pm	2009-06-03 08:41:22.000000000 +0200
@@ -220,6 +220,16 @@
         else {
             next unless -e $src;
             next if $^O eq 'cygwin' and -e "$src.exe"; # stat works, copy not
+            if (-l $src) { 
+                # check if the system supports symbolic linking 
+                use Config; 
+                if ($Config{d_symlink} && $Config{d_readlink}) { 
+                # copy as symbolic link 
+                    symlink(readlink($src), $dest); 
+                    print "$dest\n"; 
+                    next; 
+                } 
+            } 
             copy( $src, $dest ) or die "Error: couldn't copy $src to $dest: $!\n";
             print "$dest\n";
         }
