Index: trunk/config/gen/makefiles/dynpmc.in
===================================================================
--- trunk/config/gen/makefiles/dynpmc.in	(revision 44649)
+++ trunk/config/gen/makefiles/dynpmc.in	(revision 45824)
@@ -39,5 +39,4 @@
 
 PMCS_DIGEST = \
-  md2.pmc \
   md4.pmc \
   md5.pmc \
@@ -49,5 +48,4 @@
 
 OBJS_DIGEST = \
-  md2$(O) \
   md4$(O) \
   md5$(O) \
@@ -202,13 +200,4 @@
 	$(PMC2C) --library digest_group --c $(PMCS_DIGEST)
 
-md2$(O): md2.c
-	$(CC) -c @cc_o_out@md2$(O) $(INCLUDES) $(CFLAGS) md2.c
-
-md2.c: md2.dump
-	$(PMC2CC) md2.pmc
-
-md2.dump: md2.pmc
-	$(PMC2CD) md2.pmc
-
 md4$(O): md4.c
 	$(CC) -c @cc_o_out@md4$(O) $(INCLUDES) $(CFLAGS) md4.c
Index: trunk/config/gen/crypto.pm
===================================================================
--- trunk/config/gen/crypto.pm	(revision 42949)
+++ trunk/config/gen/crypto.pm	(revision 45824)
@@ -28,7 +28,4 @@
     $data{result}      = q{};
     $data{digest}      = {
-        MD2         => {
-            md_result => 'ab4f496bfb2a530b219ff33031fe06b0',
-        },
         MD4         => {
             md_result => 'd9130a8164549fe818874806e1c7014b',
